.gitlab/templates/pipeline.yaml.tpl

```
@@ -0,0 +1,297 @@
+      - .layers/datadog_extension-{{ $flavor.suffix }}/*
+  variables:
+    ARCHITECTURE: {{ $flavor.arch }}
+    SUFFIX: {{ $flavor.suffix }}
+  script:
+    - .gitlab/scripts/build_layer.sh
+
+{{ if $flavor.needs_layer_publish }}
+
+check layer size ({{ $flavor.name }}):
+  stage: build
+  image: registry.ddbuild.io/images/docker:20.10
+  tags: ["arch:amd64"]
+  needs:
+    - layer ({{ $flavor.name }})
+  dependencies:
+    - layer ({{ $flavor.name }})
+  variables:
+    LAYER_FILE: datadog_extension-{{ $flavor.suffix }}.zip
+  script:
+    - .gitlab/scripts/check_layer_size.sh
+
+{{ range $environment := (ds "environments").environments }}
+
+{{ if or (eq $environment.name "prod") }}
+
+sign layer ({{ $flavor.name }}):
+  stage: sign
+  tags: ["arch:amd64"]
+  image: ${CI_DOCKER_TARGET_IMAGE}:${CI_DOCKER_TARGET_VERSION}
+  rules:
+    - if: '$CI_COMMIT_TAG =~ /^v.*/'
+      when: manual
+  needs:
+    - layer ({{ $flavor.name }})
+    - check layer size ({{ $flavor.name }})
+  dependencies:
+    - layer ({{ $flavor.name }})
+  artifacts: # Re specify artifacts so the modified signed file is passed
+    expire_in: 1 day # Signed layers should expire after 1 day
+    paths:
+      - .layers/datadog_extension-{{ $flavor.suffix }}.zip
+  variables:
+    LAYER_FILE: datadog_extension-{{ $flavor.suffix }}.zip
+  before_script:
+    - EXTERNAL_ID_NAME={{ $environment.external_id }} ROLE_TO_ASSUME={{ $environment.role_to_assume }} AWS_ACCOUNT={{ $environment.account }} source .gitlab/scripts/get_secrets.sh
+  script:
+    - .gitlab/scripts/sign_layers.sh {{ $environment.name }}
+
+{{ end }} # if prod
+
+publish layer {{ $environment.name }} ({{ $flavor.name }}):
+  stage: publish
+  tags: ["arch:amd64"]
+  image: ${CI_DOCKER_TARGET_IMAGE}:${CI_DOCKER_TARGET_VERSION}
+  rules:
+    - if: '"{{ $environment.name }}" =~ /^(sandbox|staging)/'
+      when: manual
+      allow_failure: true
+    - if: '$CI_COMMIT_TAG =~ /^v.*/'
+
+  needs:
+{{ if eq $environment.name "prod" }}
+    - check layer size ({{ $flavor.name }})
+    - sign layer ({{ $flavor.name }})
+{{ else }}
+    - layer ({{ $flavor.name }})
+{{ end }} #end if prod
+
+  dependencies:
+{{ if or (eq $environment.name "prod") }}
+      - sign layer ({{ $flavor.name }})
+{{ else }}
+      - layer ({{ $flavor.name }})
+{{ end }} #end if prod
+
+  parallel:
+    matrix:
+      - REGION: {{ range (ds "regions").regions }}
+          - {{ .code }}
+        {{- end}}
+  variables:
+    ARCHITECTURE: {{ $flavor.arch }}
+    LAYER_FILE: datadog_extension-{{ $flavor.suffix }}.zip
+    STAGE: {{ $environment.name }}
+  before_script:
+    - EXTERNAL_ID_NAME={{ $environment.external_id }} ROLE_TO_ASSUME={{ $environment.role_to_assume }} AWS_ACCOUNT={{ $environment.account }} source .gitlab/scripts/get_secrets.sh
+  script:
+    - .gitlab/scripts/publish_layers.sh
+
+{{ if eq $environment.name "sandbox" }}
+
+publish self-monitoring sandbox layer ({{ $flavor.name }}):
```

```suggestion
publish self-monitoring sandbox layer ({{ $flavor.name }}):
```

Stage is already self-monitoring.
NIT: `publish layer sandbox [us-east-1] (...)`

---

done.